core/element_geometric_operations.h

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Owen Arnold
+#ifndef SCIPP_CORE_ELEMENT_GEOMETRIC_OPERATIONS_H
+#define SCIPP_CORE_ELEMENT_GEOMETRIC_OPERATIONS_H
+
+#include "arg_list.h"
+#include "scipp/common/overloaded.h"
+#include "scipp/core/transform.h"
+
+namespace scipp::core {
+
+/// Operators to be used with transform and transform_in_place to implement
+/// geometric operations for Variable.
+
+namespace element {
+
+constexpr auto zip_position = overloaded{
+    arg_list<float, double>,
```

Comment:
Does this compile? We only have `Eigen::Vector3d` as supported dtype. Remove `float`?

Edit: Ah, I see below you cast to double. Is there any reason for supporting float inputs? Casting is unexpected behavior, imho.

Reply:
Agreed. Will remove.